components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -499,12 +553,21 @@ public void activate(String processorName) throws AxisFault {
             if (configuration.getMessageProcessors().containsKey(processorName)) {
                 MessageProcessor processor =
                         configuration.getMessageProcessors().get(processorName);
+                CAppArtifactDataService cAppArtifactDataService = ConfigHolder.getInstance().
+                        getcAppArtifactDataService();
 
                 if (processor instanceof ScheduledMessageForwardingProcessor) {
                     MessageForwardingProcessorView view =
                             ((ScheduledMessageForwardingProcessor) processor).getView();
                     if (!view.isActive()) {
                         view.activate();
+
+                        if (!cAppArtifactDataService.isArtifactDeployedFromCApp(getTenantId(), ServiceBusConstants.MESSAGE_PROCESSOR_TYPE
```

Comment:
'ServiceBusConstants.MESSAGE_PROCESSOR_TYPE
+                                + File.separator + processorName' is used in many places. Use a variable.